tests/test_device.py

```diff
@@ -193,6 +193,24 @@ def test_apply_operation_hadamard(self, wires):
 
         assert dev.circuit == [["X", 1.0, wires], ["Y", -0.5, wires]]
 
+    @pytest.mark.parametrize("wires", [[0, 1], [1, 0], [1, 2], [2, 1]])
+    def test_operation_cnot(self, wires):

```

Comment:
Maybe it's worth also-double checking that the decomposition produces the correct matrix representation for CNOT.

Reply:
I think there is no matrix representation in `Aqt`, but I believe the decomposition is correct (found in multiple papers).